python/lib/core/dmod/core/common/reader.py

```diff
@@ -1,7 +1,20 @@
 from typing_extensions import Protocol, runtime_checkable
+from os import SEEK_SET
 
 
 @runtime_checkable
 class Reader(Protocol):
     def read(self, size: int = -1, /) -> bytes:
         """EOF if empty b''."""
+
+
+@runtime_checkable
+class Seeker(Protocol):
+    def seek(self, offset: int, whence: int = SEEK_SET) -> int:
+        """ Change the position to the given offset, returning the absolute position. """
+
+
+class ReadSeeker(Reader, Seeker):
```

Comment:
Fairly certain this needs to be changed to the following. @christophertubbs, can you verify that? 

```suggestion
@runtime_checkable
class ReadSeeker(Reader, Seeker, Protocol):
```

Reply:
Yeah, the suggested replacement looks good. Want me to hit that commit button or should we wait on Bobby?